src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -134,6 +139,9 @@ public DocumentManagementServiceImpl(RestTemplate restTemplate, SecurityUtils se
                 throw new ResourceNotFoundException(formatNotFoundMessage(documentId.toString()));
             }
         } catch (HttpClientErrorException exception) {
+            if (HttpStatus.NOT_FOUND.equals(exception.getStatusCode())) {
+                return Optional.ofNullable(responseResult).orElse(new ResponseEntity<>(HttpStatus.OK));
```

Comment:
as discussed change method return type to Optional<StoredDocumentHalResource>. And return Oprional.empty() for 404.